app/services/agreements/reports/base_report.rb

```diff
@@ -0,0 +1,27 @@
+module Agreements
+  module Reports
+    class BaseReport < ::Reports::BaseReport
+      def gen_s3_bucket_name
+        prefix = IdentityConfig.store.partner_api_bucket_prefix
+        "#{prefix}.#{ec2_data.account_id}-#{ec2_data.region}"
+      end
+
+      def save_report(report_name, body, path = nil)
+        if !IdentityConfig.store.s3_reports_enabled
+          logger.info('Not uploading report to S3, s3_reports_enabled is false')
+          return body
+        end
+        upload_file_to_s3(report_name, body, path)
+      end
+
+      def upload_file_to_s3_timestamped_and_latest(report_name, body, path)
```

Comment:
It looks like this is here to avoid uploading the timestamped files, making the name a bit of a lie. Thoughts on moving much of this into `save_report`?

It looks like the only logic we really share with the base report is the S3 uploading. How do you feel about either of these options:

1. Copy the logic for uploading to S3 here to avoid subclassing and introducing noise from the other report tooling
2. Extracting the logic for uploading to S3 into some kind of service and then using that in both of these classes, if that's the only thing that is shared

Reply:
cleaned this up.